frontend/src/metabase-lib/v1/expressions/resolver.js

```diff
@@ -58,7 +58,7 @@ const isCompatible = (a, b) => {
   }
   if (
     a === "expression" &&
-    (b === "datetime" || b === "number" || b === "string")
+    ["datetime", "number", "string", "aggregation"].includes(b)
```

Comment:
TODO for myself: consider allowing aggregations here only in the `Offset()` case